### PR TITLE
chore(cli): migrate spinner to typescript

### PR DESCRIPTION
**Pull request by Naturalclar**
## Description

Migrate `Spinner` component in gatsby-cli to typescript.

## Related Issues

#21995

### CODE REVIEW

**Comment by blainekasten** [comment]
Looks like a conflict from previous merge. Mind fixing quick?

**Comment by Naturalclar** [comment]
@blainekasten Resolved the conflict :+1:

File: packages/gatsby-cli/src/reporter/loggers/ink/components/spinner.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from "react"
+import { Box } from "ink"
+import InkSpinner from "ink-spinner"
+
+interface IProps {
+  text: string
+  statusText?: string
+}
+export function Spinner({ text, statusText }: IProps): JSX.Element {
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/spinner.tsx]:
```suggestion
export const Spinner: FunctionComponent<IProps> = ({ statusText, text }) => {
```

File: packages/gatsby-cli/src/reporter/loggers/ink/components/spinner.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from "react"
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/spinner.tsx]:
```suggestion
import React, { FunctionComponent } from "react"
```

File: packages/gatsby-cli/src/reporter/loggers/ink/components/spinner.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+interface IProps {
+  text: string
+  statusText?: string
+}
+export function Spinner({ text, statusText }: IProps): JSX.Element {
+  let label = text
+  if (statusText) {
+    label += ` — ${statusText}`
+  }
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/spinner.tsx]:
```suggestion
  const label = `${text}${statusText ? ` — ${statusText}` : ``}`
```

**Pull request merged**